### PR TITLE
Add MIX Update Subscription Element, its Parser and Serializer

### PR DESCRIPTION
**Pull request by tarun018**

### CODE REVIEW

**Comment by tarun018** [comment]
This PR is now dependent on #78 due to removal of MIXSubscribe.

File: Swiften/Elements/MIXJoin.h [hunk 1]
```diff
@@ -43,16 +43,20 @@ namespace Swift {
                 jid_ = jid;
             }
 
-            const std::vector<MIXSubscribe::ref>& getSubscriptions() const {
+            const std::unordered_set<std::string>& getSubscriptions() const {
                 return subscribeItems_;
             }
 
-            void setSubscriptions(const std::vector<MIXSubscribe::ref>& value) {
+            void setSubscriptions(const std::unordered_set<std::string>& value) {
```
Comment (tfar) [inline Swiften/Elements/MIXJoin.h]: This should be just `std::unordered_set<std::string>`. General rule, if you copy the parameter anyway, don't use `const T&`. For details see https://blogs.msdn.microsoft.com/xiangfan/2012/02/03/c98-c11-pass-by-value-or-pass-by-reference/ .

Also, the argument should be plural, so `values`.

File: Swiften/Elements/MIXJoin.h [hunk 2]
```diff
@@ -43,16 +43,20 @@ namespace Swift {
...
                 subscribeItems_ = value ;
             }
 
-            void addSubscription(MIXSubscribe::ref value) {
-                subscribeItems_.push_back(value);
+            void addSubscription(std::string value) {
+                subscribeItems_.insert(value);
+            }
+
+            bool hasSubscription(std::string value) {
```
Comment (tfar) [inline Swiften/Elements/MIXJoin.h]: This should `const std::string& value`, as you don't copy the parameter to a member or local variable. Also, this method can be const.

**Comment by tfar** [comment]
Also feel free to rebase.

File: Swiften/Elements/MIXJoin.h [hunk 3]
```diff
@@ -43,16 +43,20 @@ namespace Swift {
...
             }
 
-            const std::vector<MIXSubscribe::ref>& getSubscriptions() const {
+            const std::unordered_set<std::string>& getSubscriptions() const {
                 return subscribeItems_;
             }
 
-            void setSubscriptions(const std::vector<MIXSubscribe::ref>& value) {
-                subscribeItems_ = value ;
+            void setSubscriptions(const std::unordered_set<std::string> values) {
```
Comment (tfar) [inline Swiften/Elements/MIXJoin.h]: I think the const isn't needed anymore, as it isn't a reference anymore.

**Comment by tfar** [comment]
Thanks for the PR. It has been merged upstream.

**Pull request closed without merging**